docs/future_improvements.md

```diff
@@ -0,0 +1,104 @@
+
+This is a good solution, but it isn't without its downsides. It has the benefit of compensating the seller for their lost time, and disincentivizing bad actors, but it doesn't truly guarantee bids, doesn't provide any guarantees for other bidders, and can potentially hurt honest bidders who have simply missed the window to purchase the lot.
+
+## Future Spec (Fully-Backed Bids)
+
+The simplest solution to the bid guarantee problem on L1 is fully-backed bids - requiring bidders to lock the amount of funds they promise in a contract which can be withdrawn by the seller when the bidder wins. To increase his bid, the bidder has to increase his bid again with more funds from his wallet. 
+
+On L2, this scheme runs into problems when the flow between the user's wallet on the L1 and their bid on the L2 isn't fast or consistently available. We decided against fully backed bids in the initial spec because at the time funds could only be moved between L1 and L2 when the Hydra head first starts, clearly breaking both speed and constancy requirements.
+
+Increasing bids on L2 requires incremental commit/decommit, a feature which has recently been rolled out by the Hydra team. Unfortunately, according to our analysis, that feature is not yet mature enough for integration into our development timeline. As such, we have decided to leave Fully-Backed Bids un-implemented.
+
+### Auction Flow
+
+The auction logic itself greatly simplifies when designing for fully backed bids by including the amount of ada specified in the bid in the standing bid utxo, and enforcing the refunding of the previous bidder when submitting a tx with the `NewBid` redeemer.
+
+With these changes we are able to do without deposits, the reclaim period, or any extra cleanup transactions, and able to discard the many various datum fields and redeemers associated with them.
+
+The simplified auction flow looks like this:
+
+```mermaid
+stateDiagram-v2
+  [*]           -->   Announced:     AnnounceAuction [P0]
+  Announced     -->   BiddingOnL1:   StartBidding [P2]
+  BiddingOnL1   -->   BiddingOnL2:   CommitStandingBid [P2]
+  BiddingOnL1   -->   Purchase:      [P3]
+  BiddingOnL2   -->   BiddingOnL1:   FanoutStandingBid [P2]
+  BiddingOnL2   -->   Purchase:      FanoutStandingBid [P3]
+  Purchase      -->   [*]:           Conclude [P4]
+```
+
+There are two changed state transitions:
+
+<table><tr><td>
+
+</td></tr><tr></tr><tr><td>
+
+`CommitStandingBid [P2]`. During the bidding period the standing bid can be moved to the
+ Hydra Head, allowing bidders to place bids in the auction via L2 transactions (but no longer 
+ via L1). This is no longer dependent on the Head Initializing (HI) state, as the commit can be made with the head still running.
+
+</td></tr><tr></tr><tr><td>
+
+`Resolve [P4]`. During the purchase period, anyone can spend the standing bid utxo, sending the lot to the winning bidder and the deposit to the seller, while also recovering the min 2 ADA inside it.
```

Comment:
Should be `Conclude` I guess.

Reply:
I went with Resolve instead of Conclude for this spec because it encapsulates the actions taken in the Claim & Conclude parts of the deposit-based auction. I see now that I missed the rename in the state diagram so I'll rename it there.

Good catch, thanks!